### PR TITLE
Better dependencies cache for CI

### PR DESCRIPTION
**Pull request by shekhirin**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -23,13 +23,7 @@ jobs:
...
     - uses: actions/checkout@v2
     - name: Cache dependencies
-      uses: actions/cache@v2
-      with:
-        path: |
-          ~/.cargo
-          ./Cargo.lock
-          ./target
-        key: ${{ matrix.os }}-${{ matrix.rust }}-${{ hashFiles('Cargo.toml') }}
+      uses: Swatinem/rust-cache@c5ed9ba6b7e1bb8aff90d43acd2f0af4990fa57c
```
Comment (curquiza) [inline .github/workflows/rust.yml:26]:
```suggestion
      uses: Swatinem/rust-cache@v1
```

It is possible to use `@v1` instead? :)

Comment (shekhirin) [reply .github/workflows/rust.yml:26]: I'd not go for it 😄 

> I use the v1 branch similar to master development, so if you want to have a more stable experience, please use a fixed revision or tag.

(https://github.com/Swatinem/rust-cache#versioning)

**Comment by curquiza** [comment]
Hello @shekhirin! Thanks for your PR 🙂

What is the difference with the previous cache system we used in our CIs?

File: .github/workflows/rust.yml (hunk 1 above)
Comment (curquiza) [reply .github/workflows/rust.yml]: Can we use a version then, I mean a more human readable version, like v1.3.0? :)

**Comment by shekhirin** [comment]
Hey @curquiza,

It caches almost the same directories (https://github.com/Swatinem/rust-cache/blob/c5ed9ba6b7e1bb8aff90d43acd2f0af4990fa57c/src/common.ts#L76-L88), main difference is the clean up in `target` directory and some of `~/.cargo` subdirectories before caching (https://github.com/Swatinem/rust-cache/blob/v1/src/save.ts#L35-L49).

File: .github/workflows/rust.yml (hunk 1 above)
Comment (shekhirin) [reply .github/workflows/rust.yml]: yep, done!

**Comment by curquiza** [comment]
Thanks for your explanation @shekhirin! :)

I put @irevoire as a final reviewer, as he was the first one integrating the cache to the CI :)

**Review by irevoire — APPROVED** [review]
It looks good to me!
It's faster and way more concise, once again thanks @shekhirin :tada: 

bors merge

**Pull request merged**